addons/im_livechat/models/res_partner.py

```diff
@@ -79,3 +96,11 @@ def _compute_display_name(self):
             return
         for partner in self:
             partner.display_name = partner.name
+
+    def action_view_livechat_sessions(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('im_livechat.discuss_channel_action')
+        action['domain'] = expression.AND([
+            [('channel_type', '=', 'livechat')],
+            [('channel_member_history_ids.partner_id', 'in', self.ids)]
+        ])
```

Comment:
I would not overwrite the domain totally (the live chat part is already taken care of) but instead append the new condition to the existing domain.